pkg/cmd/membership/membership.go

```diff
@@ -0,0 +1,168 @@
+package membership
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/pkg/transport"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+
+	"github.com/openshift/cluster-etcd-operator/pkg/cmd/waitforkube"
+)
+
+const (
+	retryDuration     = 2 * time.Second
+	dialTimeout       = 5 * time.Second
+	etcdCertFile      = "/var/run/secrets/etcd-client/tls.crt"
+	etcdKeyFile       = "/var/run/secrets/etcd-client/tls.key"
+	etcdTrustedCAFile = "/var/run/configmaps/etcd-ca/ca-bundle.crt"
+	dbPath            = "member/snap/db"
+)
+
+type membership struct {
+	errOut         io.Writer
+	EtcdDnsName    string
+	EtcdEndpoints  string
+	etcdDataDir    string
+	EtcdDataFile   string
+	ExistingMember bool
+}
+
+// NewWaitForKubeCommand waits for kube to come up before continuing to start the rest of the containers.
+func NewMembershipCommand(errOut io.Writer) *cobra.Command {
+	membership := &membership{
+		errOut: errOut,
+	}
+	cmd := &cobra.Command{
+		Use:   "membership",
+		Short: "cluster membership validate",
+		Long:  "This command makes sure that the local etcd member is part of the cluster.",
+		Run: func(cmd *cobra.Command, args []string) {
+			must := func(fn func() error) {
+				if err := fn(); err != nil {
+					if cmd.HasParent() {
+						klog.Fatal(err)
+						fmt.Fprint(membership.errOut, err.Error())
+					}
+				}
+			}
+			must(membership.Complete)
+			must(membership.Validate)
+			must(membership.Run)
+		},
+	}
+	membership.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (m *membership) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&m.EtcdDnsName, "etcd-dns-name", "", "etcd member FQDN")
+	fs.StringVar(&m.EtcdEndpoints, "endpoints", "", "commas seperated list of etcd clients")
+	fs.StringVar(&m.etcdDataDir, "data-dir", "/var/lib/etcd", "path to etcd data-dir")
+}
+func (m *membership) Validate() error {
+	if m.IsExistingMember() {
+		if m.EtcdEndpoints == "" || m.EtcdDnsName == "" {
+			return fmt.Errorf("membership validation requires etcd-dns-name and endpoints")
+		}
+	}
+	return nil
+}
+
+func (m *membership) Complete() error {
+	m.EtcdDataFile = filepath.Join(m.etcdDataDir, dbPath)
+
+	etcdDnsName, etcdEndpoints := getENV()
```

Comment:
why does `getENV` have to be a separate function, replace it with `x, y := os.Getenv("x"), os.Getenv("y")` construct?

Reply:
just readability

Reply:
We are essentially just doing this https://github.com/kubernetes/client-go/blob/master/rest/config.go#L456

Reply:
In my opinion, I think it is more readable if it is `x, y := os.Getenv("x"), os.Getenv("y")` than as a separate function, unless we intend to do more in that function in future.